src/SurveyDesign.jl

```diff
@@ -30,53 +29,93 @@ The population size is equal to the sample size unless `popsize` is explicitly p
+                end
+                weights = 1 ./ probs
+            elseif isa(probs, Symbol)
+                probs = data[!, probs]
+                if !all(y -> y == first(probs), probs) # SRS by definition is equi-weighted
+                    error("Simple Random Sample must be equi-weighted. Different sampling probabilities detected in vector")
+                end
+                weights = 1 ./ probs
+            end
+            # If all weights are equal then estimate
+            equal_weight = first(weights)
+            popsize = round(sampsize .* equal_weight) |> UInt
+        elseif typeof(popsize) <: Vector{<:Real}
+            if !all(y -> y == first(popsize), popsize) # SRS by definition is equi-weighted
+                error("Simple Random Sample must be equi-weighted. Different sampling weights detected in vectors")
+            end
+            weights = popsize ./ sampsize
+            popsize = first(popsize) |> UInt
+        else
+            error("If popsize not given then either sampling weights or sampling probabilities must be given")
         end
-        # add frequency weights column to `data`
-        data[!, :weights] = weights
-        # add probability weights column to `data`
-        data[!, :probs] = 1 ./ data[!, :weights]
         # set sampling fraction
-        sampfraction = sampsize / popsize
+        sampfraction = sampsize ./ popsize
         # set fpc
-        fpc = ignorefpc ? 1 : 1 - (sampsize / popsize)
-
+        fpc = ignorefpc ? 1 : 1 .- (sampsize ./ popsize)
+        # Add columns for weights and probs in data -- Check if really needed to add them as columns
+        if !isnothing(probs)
+            # add probability weights column to `data`
+            data[!, :probs] = probs
+            # add frequency weights column to `data`
+            data[!, :weights] = 1 ./ data[!, :probs]
+        else # else weights were specified
+            # add frequency weights column to `data`
+            data[!, :weights] = weights
+            # add probability weights column to `data`
+            data[!, :probs] = 1 ./ data[!, :weights]
+        end
         new(data, sampsize, popsize, sampfraction, fpc, ignorefpc)
     end
 end
 
 # `show` method for printing information about a `SimpleRandomSample` after construction
 function Base.show(io::IO, ::MIME"text/plain", design::SimpleRandomSample)
-    printstyled("Simple Random Sample:\n"; bold = true)
-    printstyled("data: "; bold = true)
+    printstyled("Simple Random Sample:\n"; bold=true)
+    printstyled("data: "; bold=true)
     print(size(design.data, 1), "x", size(design.data, 2), " DataFrame")
-    printstyled("\nweights: "; bold = true)
+    printstyled("\nweights: "; bold=true)
     print_short(design.data.weights)
-    printstyled("\nprobs: "; bold = true)
+    printstyled("\nprobs: "; bold=true)
     print_short(design.data.probs)
-    printstyled("\nfpc: "; bold = true)
+    printstyled("\nfpc: "; bold=true)
```

Comment:
Maybe have a check here. 

```julia
printstyled("\nignore fpc: "; bold=true)
print(design.ignorefpc)

if (!design.ignorefpc)
    printstyled("\nfpc: "; bold=true)
    print_short(design.fpc)
end
```

Reply:
No, if ignorefpc is true, then fpc should print as `1`, just as sanity check